### PR TITLE
Fix #6622 bad Title.path in ML sites if parent slug is created or mod…

### LINKED ISSUE #6622: The slugs of child pages are built incorrectly and return a 404 error when a user adds a custom slug for another language of the parent page
### Summary

The slugs of child pages are breaking when a user adds a custom slug for another language of the parent page, this leads to incorrect urls being displayed and the correct urls return a 404 error.

### Actual behaviour

1. Create an empty website with 2 languages - English and German.
2. Create a Page with the slug `parent-en` in English and publish it, leave the German slug empty.
3. Create a child Page of the first page with the slug `child-en` in English and `child-de` in German
4. Confirm that the created child page has the url `parent-en/child-en` and the respective German url, all is fine for now.
5. Open the parent Page and add a custom German slug, specifically `parent-de` and publish all the unpublished pages.
6. Now open the German version of the website, and see in the menu that the link to the child page is `parent-en/child-de` instead of the correct `parent-de/child-de` url. Confirm that the expected url `parent-de/child-de` actually returns a 404 error.

### Expected behaviour

On the 6th the url in the menu should be `parent-de/child-de` , and the url `parent-de/child-de` shouldn't return a 404 error.

### Environment

* the first one
    * Python version: 3.5.3
    * Django version: 1.11.20
    * django CMS version: 3.6.0

* the second one
    * Python version: 2.7.6
    * Django version: 1.9.5
    * django CMS version: 3.5.1

The issue doesn't appear to be present in django-cms 3.4.5 though.

### PR DESCRIPTION
**Pull request by fp4code**
…ified

## Description

Fix #6622 Bad Title.path in Multilanguage sites if parent slug is created or modified

### First bug:

In 'en','fr' site
1) Create /en/en-1/ page
2) Create from this page a subpage with slug "en-1-1"
3) Create "fr" translation with slug "fr-1-1"
4) Create "fr" translation of parent /en/en-1/ with slug "fr-1"

As a result, the path of subpage "fr-1-1" is not updated. Here the paths:

/en/en-1/
/en/en-1/en-1-1/
/fr/fr-1/
/fr/en-1/fr-1-1/   <-- Bad Title.path "en-1/fr-1-1" instead of "fr-1/fr-1-1"

### Second bug:

Change "en" slug of /en/en-1/ page to "en-1b",
As a result the path of subpage "en-1-1" is changed to "fr-1/en-1-1" instead of "en-1b/en-1-1". Here the paths:

/en/en-1b/
/en/fr-1/en-1-1/   <-- Bad Title.path "fr-1/en-1-1" instead of "en-1b/fr-1-1"
/fr/fr-1/
/fr/en-1/fr-1-1/   <-- Bad Title.path "en-1/fr-1-1" instead of "fr-1/fr-1-1"

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have updated the **CHANGELOG.rst**
* [x] I have added or modified the tests when changing logic (no changed logic)

### CODE REVIEW

**Review by Aiky30 — CHANGES_REQUESTED** [review]
This change will need a test that replicates and enforces the business requirements that have been identified. Without tests we cannot guarantee the fix in the future.

Once a test is implemented and approved we can look to get it released to unblock you ASAP.

**Comment by fp4code** [comment]
Sorry for the mess, I don't know how to combine PR.
#6965 is the fix of #6622
#6967 add a test file with no fix to observe the issue #6622
#6968 the fix with test file included.

**Pull request closed without merging**
